storage/store.go

```diff
@@ -647,12 +647,14 @@ func (s *Store) MaybeGossipConfigs() error {
 		if _, err := rng.getLeaseForGossip(s.Context(nil)); err != nil {
 			return err
 		}
-		// Always gossip the configs as there are leader lease acquisition
-		// scenarios in which the config is not gossiped (e.g. leader executes
-		// but crashes before gossiping; comes back up but group goes dormant).
-		rng.maybeGossipConfigs(func(configPrefix proto.Key) bool {
-			return bytes.Equal(configPrefix, cd.keyPrefix)
-		})
+		// Configs are gossiped immediately when forced. Otherwise, configs
+		// will be gossiped when they are updated or a leader lease is
+		// acquired/extended.
+		if force {
+			rng.maybeGossipConfigs(func(configPrefix proto.Key) bool {
+				return bytes.Equal(configPrefix, cd.keyPrefix)
+			})
+		}
```

Comment:
Sure, the problem was that configs were never gossiped if a lease is acquired during bootstrap.

1) `BootstrapCluster` creates a new store without `Gossip`. (A `StoreContext` created at Line 108 of node.go doesn't have a `Gossip`.
2) The range acquires a leader lease but does not gossip configs since `r.rm.Gossip()` returns `nil`.
3) `Node.start` is called. Gossiping starts.
4) Before any update to configs or a leader lease is extended, the range is split. 
5) `updateRangeInfo` is called to get zone max bytes and fails since there is no gossiped configs.

`Example_ranges` in `server/cli/cli_test.go` hit this bug.

@bdarnell , will try changing `NewRange` to call `Range.maybeGossipConfigs`.